src/Summoner/Template.hs

```diff
@@ -109,6 +111,34 @@ createStackTemplate ProjectData{..} = Dir (toString repo) $
         $endLine
         |]
 
+    ghcOptions :: Text
+    ghcOptions = case warnings of
+        [] -> extraWarnings
+        xs -> T.intercalate "\n" xs
+
+    extraWarnings :: Text
+    extraWarnings =
+        [text|
+        -Wincomplete-uni-patterns
+        -Wincomplete-record-updates
+        -Wmissing-import-lists
+        -Wcompat
+        -Widentities
+        $versionWarnings
+        |]
+
+    versionWarnings :: Text
+    versionWarnings
+        =  memptyIfFalse (Ghc7103 `notElem` testedVersions)
+            "-Wredundant-constraints\n"
+        <> memptyIfFalse (null (testedVersions \\ [Ghc822, Ghc843]))
```

Comment:
This implementation doesn't scale. Imagine that GHC-8.6.1 released. And now we need not forget to update 
 all such places. It's recommended to write code in a such way that minimizes amount of things needed to be remembered. It's better if you implement function like:

```haskell
hasAtLeast :: Ord a => [a] -> a -> Bool
```

That checks that there's at least one element that greater than given element.

And then you can write only once:

```haskell
meptyIfFalse (testedVersions `hasAtLeast` Ghc822)
    "some text"
```